lib/gpu/lib/src/buffer.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:nativewrappers';
+import 'dart:typed_data';
+
+import 'context.dart';
+
+/// A buffer data range.
+class BufferView {
+  /// The buffer of this view.
+  final Buffer buffer;
+  /// The start of the view, in bytes starting from the beginning of the
+  /// [buffer].
+  final int offsetInBytes;
+  /// The end of the view
+  final int lengthInBytes;
+
+  /// Create a new view into a buffer on the GPU.
+  const BufferView(
+      {required this.buffer,
+      required this.offsetInBytes,
+      required this.lengthInBytes});
+}
+
+/// A buffer that can be referenced by commands on the GPU.
+mixin Buffer {}
```

Comment:
Generally speaking, a completely empty interface is a pretty big code smell.

What other kinds of buffers are we going to have?  Are device buffers going to be usable in the same context as host buffers?

Reply:
Yeah, `DeviceBuffer` is the other kind of buffer we need to support, and they will always be usable in the same context as `HostBuffer`.

Reply:
I don't really know if this is how the final API is going to end up. My guess is that I'll work out something better when we add actual commands.

Reply:
If you don't need it yet then don't add the interface yet.

Reply:
Done.